code/create/dta_dicom2spec_rules.py

```diff
@@ -0,0 +1,226 @@
+                               "phase1", "phase2", "magnitude1", "magnitude2",
+                               "fieldmap", "epi", "meg"]
+
+        for m in direct_search_terms:
+            if m in prot_parts:
+                return m
+            if "mprage" in prot_parts:
+                return "t1w"
+
+        # BEGIN: Additional rule for forrest-structural
+        # TODO: Probably to be moved to some rule enhancement
+        if "st1w" in prot_parts:
+            return "t1w"
+        if "st2w" in prot_parts:
+            return "t2w"
+        if "tof" in prot_parts:
+            return "angio"
+        # END
+
+        if "func" in prot_parts:
+            return "bold"
+
+    # found nothing, but modality isn't necessarily required
+    return None
+
+def _guess_task(dcm_dict, record):
+    protocol = record.get("ProtocolName", None)
+    count_if_same_protocol_but_scanned_earlier = 1
+    #
+    exp_flag = False
+
+    if protocol:
+        if protocol.lower() in "dta_epi_pmu_rest":
+               return "rest"
+        if protocol.lower() in "DTA_ep2d_bold_Task".lower():
+            for elem in dcm_dict._dicom_series:
+               if "DTA_ep2d_bold_Task".lower() in elem.get("ProtocolName").lower():
+                  return ( "exp" + get_column( record.get("PatientID") ) )
+
+        if "60dir".lower() in protocol.lower():
+            return "60dir"
+        if "120dir".lower() in protocol.lower():
+            return "120dir"
+
+
+        import re
+        prot_parts = re.split('_|-|\s', protocol.lower())
+        try:
+            idx = prot_parts.index("task")
+            task = prot_parts[idx + 1]
+            return task
+        except (ValueError, IndexError):
+            # default to entire protocol name?
+            # This should actually check the results of other guesses
+            # (like modality) to determine a better default than nothing.
+            # At least parts of the protocol name that were already matched elsewhere
+            # should be excluded
+            return None
+    else:
+        # default to entire protocol name?
+        # see above
+        return None
+
+
+
+def _guess_run(dcm_dict, record):
+        run = 1
+        # vielleicht andere Variablen-Namen?
+        protocol = record.get("ProtocolName")
+        # the bold-tasks don't have runs.
+        if "DTA_ep2d_bold_Task".lower() in protocol.lower():
+              return "1"
+        if "DTA_epi_pmu_rest".lower() in protocol.lower():
+              return "1"
+
+        if "60dir".lower() in protocol.lower():
+            return "60dir"
+        if "120dir".lower() in protocol.lower():
+            return "120dir"
+
+
+#        if "DTA_fieldmap_B0_gre_t2star".lower() in protocol.lower():
+#              return "1" # here sollte nur ne Nummer sein, nicht das "run-"?
+        if protocol:
+            for elem in dcm_dict._dicom_series:
+                if protocol in str(elem): ## change this..
```

Comment:
Similarly to above I don't quite get, what's the idea here. We should talk about it.